dns_discover.go

```diff
@@ -40,6 +41,20 @@ func discoverDNS(domain string, port int) (servers []string, ttl time.Duration,
 	return
 }
 
+// retryingFindTXT will, on any DNS failure, retry for up to 10 minutes before
```

Comment:
Super minor point, but the default `backoff.NewExponentialBackoff()` has a max time of 15 minutes, not 10. Or is 10 minutes coming from somewhere else?